meta-intel-edison-distro/recipes-core/post-install/files/post-install.sh

```diff
@@ -31,20 +31,40 @@ sshd_init () {
     systemctl start sshdgenkeys
 }
 
+# Substitute the SSID and passphrase in the file /etc/hostapd.conf
+# The SSID is built from the hostname and a serial number to have a
+# unique SSID in case of multiple Edison boards having their WLAN AP active.
+setup_ap_ssid_and_passphrase () {
+    # factory_serial is 16 bytes long
+    if [ -f /sys/class/net/wlan0/address ];
+    then
+        wlan0_addr=$(cat /sys/class/net/wlan0/address | tr '[:lower:]' '[:upper:]')
+        ssid="EDISON-${wlan0_addr:12:2}-${wlan0_addr:15:2}"
+    fi
 
-# script main part
+    if [ -f /factory/serial_number ] ;
```

Comment:
This one IIRC was populated by one of the first-boot scripts, so theoretically may not exist everywhere or may be deleted by the user. If there's no such script in place right now (to guarantee the serial will always exist), it's probably worth it to use some fallback values for the SSID and password to prevent  creating invalid or passwordless config.

Reply:
Sure? The directory contains 2 files with serial number equal to the label on the edison and the other a unique bluetooth address. I was under the impression the files are created at production time and never overwritten during flashing (or I would have lost them long ago). Nevertheless it is good advice to have some defaults just in case. I'll have a look.

Reply:
I think [this function](https://github.com/edison-fw/meta-intel-edison/commit/b38c0fec27797b906593091fcb26badb6c526758#diff-6ae83b99829fe64a9082facc28b8538bL70) is the source for my impression, but I can't find any code that actually populates those files with meaningful data, so probably that dummy data in the function is a fallback similar to what I've proposed in this comment.

Reply:
Ah, I forget about that one. Yes, I didn't find flashall writing that factory partion either. I guess in the factory they had a netboot over rndis or cdc, generate serial and pick bd_addr from database, then put label with serial. But once you loose the partition, there is no way to recover it.

Reply:
I had a closer look at the original code, and as I understand now on first run /factory is created and mounted, only if the mount fails a partition is created and defaults are entered. As the code says, that can happen in the factory. But why don't we now error out? We are not creating /factory anymore. Is this taken care of by systemd due to our line in fstab:
```
/dev/disk/by-partlabel/factory  /factory        auto    noauto,x-systemd.automount,nosuid,nodev,noatime,discard     1   1
```

Reply:
Maybe that's because none of the systems anyone used have actually lost their partitions and the reflashing process being used doesn't touch the existing ones? Alternatively - people haven't noticed :)

Reply:
I meant to say: we produce a new root image, afaik without /factory directory. Still we are able to mount the factory partition onto it. So, who creates the /factory directory?